test/test_parser.cpp

```diff
@@ -724,6 +724,19 @@ TEST_TYPES(Parser_Numerics, Prop<Int>, Nullable<Int>, Indexed<Int>, NullableInde
         verify_query_sub(test_context, t, util::format("values == $%1", i), args, 1);
     }
     verify_query(test_context, t, "values == null", nullable ? 1 : 0);
+    verify_query(test_context, t, "values == ANY {-1, 0, 1}", 3);
+    verify_query(test_context, t, "values == NONE {-1, 0, 1}", t->size() - 3);
+    verify_query(test_context, t, "values == NONE {-1, 0}", t->size() - 2);
+    verify_query(test_context, t, "values == NONE {-1}", t->size() - 1);
+    verify_query(test_context, t, "values == NONE {}", t->size());
+    verify_query(test_context, t, "values != ALL {-1, 0, 1}", t->size() - 3);
+    verify_query(test_context, t, "values != ALL {-1, 0}", t->size() - 2);
+    verify_query(test_context, t, "values != ALL {-1}", t->size() - 1);
+    verify_query(test_context, t, "values != ALL {}", t->size());
+    verify_query(test_context, t, "values == ALL {-1, 0, 1}", 0);
+    verify_query(test_context, t, "values == ALL {-1, 0}", 0);
+    verify_query(test_context, t, "values == ALL {-1}", 1);
+    verify_query(test_context, t, "values == ALL {}", t->size());
```

Comment:
how about a few more? like: 
```
+    size_t sz = t->size();
+
+    verify_query(test_context, t, "values == ANY {0, 1}", 2);
+    verify_query(test_context, t, "values == ANY {1}", 1);
+    verify_query(test_context, t, "values == ANY {}", 0);
+
+    verify_query(test_context, t, "values != NONE {-1, 0, 1}", 3);
+    verify_query(test_context, t, "values != NONE {-1, 0}", 2);
+    verify_query(test_context, t, "values != NONE {-1}", 1);
+    verify_query(test_context, t, "values != NONE {}", 0);
+
+    verify_query(test_context, t, "values != ANY {-1, 0, 1}", sz - 3);
+    verify_query(test_context, t, "values != ANY {0, 1}", sz - 2);
+    verify_query(test_context, t, "values != ANY {1}", sz - 1);
+    verify_query(test_context, t, "values != ANY {}", sz);
```

not all this makes sense, but would be good to clarify. Surprisingly results are not that above.

Reply:
Thanks, tests added. The `!= NONE {...}` are difficult to reason about what the right results should be because of the double negative, but I think we can simplify by rewriting those as `== ALL {...}` and indeed the results are consistent with those positive queries.